implementation/core/src/main/java/io/smallrye/faulttolerance/core/metrics/GeneralMetricsEvents.java

```diff
@@ -0,0 +1,16 @@
+package io.smallrye.faulttolerance.core.metrics;
+
+import io.smallrye.faulttolerance.core.InvocationContextEvent;
+
+public class GeneralMetricsEvents {
+    public static class ExecutionFinished implements InvocationContextEvent {
```

Comment:
again, seems to me that an enum would be a better fit

Reply:
Enum should actually work well here, I'll check.